iceoryx_hoofs/source/posix_wrapper/semaphore.cpp

```diff
@@ -41,14 +39,16 @@ Semaphore& Semaphore::operator=(Semaphore&& rhs) noexcept
 
         CreationPattern_t::operator=(std::move(rhs));
 
-        m_name = std::move(rhs.m_name);
-        m_isCreated = std::move(rhs.m_isCreated);
-        m_isNamedSemaphore = std::move(rhs.m_isNamedSemaphore);
-        m_handle = std::move(rhs.m_handle);
-        m_isShared = std::move(rhs.m_isShared);
+        // for details see here iceoryx_hoofs/include/iceoryx_hoofs/design_pattern/creation.hpp
+        m_name = std::move(rhs.m_name); // NOLINT(hicpp-invalid-access-moved)
+        // the rest are trivially-copyable, so no std::move
+        m_isCreated = rhs.m_isCreated;
+        m_isNamedSemaphore = rhs.m_isNamedSemaphore;
+        m_handle = rhs.m_handle;
```

Comment:
Here you have to use move since the `m_handle` type is in mac os a unique_ptr like structure which can only be moved and not copied.
This is a trade off of platform independent code.

Reply:
done